packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,22 +1,67 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
 
 export type HeaderProps = {
+	productId: number;
 	title: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+const DEFAULT_PRODUCT_NAME = 'AUTO-DRAFT';
+
+export function Header( { productId, title }: HeaderProps ) {
+	const { isProductLocked, isSaving } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord } = select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				isProductLocked: isPostSavingLocked(),
+				isSaving: isSavingEntityRecord(
+					'postType',
+					'product',
+					productId
+				),
+			};
+		},
+		[ productId ]
+	);
+
+	const isDisabled = isProductLocked || isSaving;
+	const isCreating = title === DEFAULT_PRODUCT_NAME;
+
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	function handleSave() {
+		saveEditedEntityRecord( 'postType', 'product', productId );
```

Comment:
We should also re-direct to the new product form when saving a new post for the first time.
But I do realize that might of been out of scope for this issue.

Reply:
see -> [64df88c](https://github.com/woocommerce/woocommerce/pull/37152/commits/64df88c7944eaebd1f34f7ea380bf216be0d9b45)
see -> [0665be4](https://github.com/woocommerce/woocommerce/pull/37152/commits/0665be4d19fc2bc3e301c3ee4ceb7a501e0d3ae9)